src/Core/Config.cpp

```diff
@@ -160,6 +160,12 @@ bool Config::IsPeerBlocked (const std::string peer) {
 	return false;
 }
 
+bool Config::IsPeerPreferred (const std::string peer) {
+	std::vector<std::string> vec = m_pImpl->m_nodeConfig.GetP2P().GetPreferredPeers();
```

Comment:
You need to store this variable as a `const std::vector<std::string>&` , or else this will be doing an expensive copy on every single usage.

Also, since you're always treating preferred peers as a set, you should be storing it as one (actually, a `std::unordered_set`, which tends to have an even faster lookup time)